src/common/log/log.hpp

```diff
@@ -0,0 +1,249 @@
+/*
+ * Copyright (C) 2023 Renesas Electronics Corporation.
+ * Copyright (C) 2023 EPAM Systems, Inc.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#ifndef LOG_COMMON_HPP_
+#define LOG_COMMON_HPP_
+
+#include <cstdio>
+#include <cstring>
+
+#include "config.hpp"
+#include "utils/stringer.hpp"
+#include "utils/utils.hpp"
+
+/**
+ * Helper macro to display debug log.
+ */
+#if CONFIG_LOG_LEVEL >= CONFIG_LOG_LEVEL_DEBUG
+#define LOG_MODULE_DBG(module) aos::Log(module, LogLevelType::eDebug)
+#else
+#define LOG_MODULE_DBG(module) true ? (void)0 : aos::LogVoid() & aos::Log(module, LogLevelType::eDebug)
```

Comment:
It is a trick. To make log line not compiled at all. With (void)0 it is not possible because `(void)0 << "some string";` is not compiled but: `true ? (void)0 : aos::LogVoid() & aos::Log(module, LogLevelType::eDebug) << "some string";` compiles and not included to the final binary.